compose/foundation/foundation/src/desktopTest/kotlin/androidx/compose/foundation/gestures/DragGestureTest.kt

```diff
@@ -0,0 +1,287 @@
+            assertFalse(dragEnded)
+
+            scene.sendPointerEvent(
+                PointerEventType.Move,
+                Offset(5f + viewConfiguration.touchSlop, 15f)
+            )
+            assertEquals(0f, dragOffset.x)
+            assertEquals(10f, dragOffset.y)
+            assertEquals(2, onDragCounter)
+            assertFalse(dragCanceled)
+            assertFalse(dragEnded)
+
+            scene.sendPointerEvent(
+                PointerEventType.Move,
+                Offset(5f + viewConfiguration.touchSlop, 25f)
+            )
+            assertEquals(0f, dragOffset.x)
+            assertEquals(10f, dragOffset.y)
+            assertEquals(3, onDragCounter)
+            assertFalse(dragCanceled)
+            assertFalse(dragEnded)
+
+            scene.sendPointerEvent(
+                eventType = PointerEventType.Release,
+                position = Offset(viewConfiguration.touchSlop, 15f),
+                button = PointerButton.Primary,
+            )
+            assertEquals(-5f, dragOffset.x)
+            assertEquals(-10f, dragOffset.y)
+            assertTrue(dragEnded)
+            assertFalse(dragCanceled)
+            assertEquals(4, onDragCounter)
+        }
+    }
+
+    @Test
+    fun `draggable by touch`() {
+        val density = Density(1f)
+        val viewConfiguration = DefaultViewConfiguration(density)
+
+        ImageComposeScene(
+            width = 100,
+            height = 100,
+            density = density
+        ).use { scene ->
+
+            var dragStartResult: Offset? = null
+            var dragCanceled = false
+            var dragEnded = false
+            var onDragCounter = 0
+            var dragOffset = Offset.Zero
+
+            scene.setContent {
+                Box(
+                    modifier = Modifier
+                        .size(40.dp, 40.dp)
+                        .onDrag(
+                            enabled = true,
+                            matcher = PointerMatcher.touch,
+                            onDragStart = { offset -> dragStartResult = offset },
+                            onDragCancel = { dragCanceled = true },
+                            onDragEnd = { dragEnded = true },
+                            onDrag = {
+                                dragOffset = it
+                                onDragCounter++
+                            }
+                        )
+                )
+            }
+
+            // Note: touch slop is different from mouse slop,
+            // so values a bit different in this test comparing with mouse drag
+            scene.sendPointerEvent(PointerEventType.Move, Offset(5f, 5f), type = PointerType.Touch)
+            scene.sendPointerEvent(PointerEventType.Press, Offset(5f, 5f), type = PointerType.Touch)
+            scene.sendPointerEvent(
+                PointerEventType.Move,
+                Offset(5f + viewConfiguration.touchSlop + 5f, 5f),
+                type = PointerType.Touch
```

Comment:
I added a test to check that when onDrag expects touch events only, then drag will not start if the gesture starts using the mouse.

But, at the moment we can't have a test to check that onDrag(matcher=touch) never reacts on mouse, because `pointerInputEvent()` always creates a list of changes with hardcoded pointerId = 0. Therefore drag will keep running regardless of whether it's mouse or touch. 

But we'll add such a test on android when upstreaming this, or when we support touch/stylus inputs on desktop.

Reply:
> I added a test to check that when onDrag expects touch events only, then drag will not start if the gesture starts using the mouse.

Yes, I meant this test, not the test when we start a drag with touch, and continue the drag with mouse.